src/internal/connector/graph/concurrency_limiter.go

```diff
@@ -0,0 +1,52 @@
+package graph
+
+import (
+	"net/http"
+	"sync"
+
+	khttp "github.com/microsoft/kiota-http-go"
+)
+
+// ConcurrencyLimiter middleware limits the number of concurrent requests to graph API.
+// Currently we only enforce concurrency limit for exchange
+type ConcurrencyLimiter struct {
+	semaphoreExch chan struct{}
+}
+
+var (
+	once               sync.Once
+	concurrencyLimiter *ConcurrencyLimiter
+	// Outlooks expects max 4 concurrent requests
+	// https://learn.microsoft.com/en-us/graph/throttling-limits#outlook-service-limits
+	maxConcurrentRequestsExchange = 4
```

Comment:
Not much point in calling out exchange here unless you're going to take in a service identifier as part of the `initialize` func.  Otherwise the correlation between the two is purely in cognitive load, and not in code.